### PR TITLE
[FLINK-21152][build] Bump flink-shaded to 13.0 

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/FlinkDeserializationContext.java [hunk 1]
```diff
@@ -56,6 +56,13 @@ protected FlinkDeserializationContext(
         this.objectMapper = src.objectMapper;
     }
 
+    protected FlinkDeserializationContext(
```
Comment (fapaul) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/FlinkDeserializationContext.java:59]: How did the version bump affect these lines?

File: pom.xml [hunk 2]
```diff
@@ -122,7 +122,7 @@ under the License.
 		<zookeeper.version>3.4.14</zookeeper.version>
 		<!-- Only the curator2 TestingServer works with ZK 3.4 -->
 		<curator.version>2.12.0</curator.version>
-		<jackson.version>2.10.1</jackson.version>
+		<jackson.version>2.12.1</jackson.version>
```
Comment (fapaul) [inline pom.xml:125]: Is this always updated when the jackson version in flink-shaded is updated?

**Review by fapaul — APPROVED** [review]
Overall LGTM, I left some minor questions for my understanding.

File: pom.xml (hunk 2 above)
Comment (zentol) [reply pom.xml:125]: yes, this effectively is the flink-shaded-jackson version.

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/FlinkDeserializationContext.java (hunk 1 above)
Comment (zentol) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/FlinkDeserializationContext.java:59]: The parent class comes from jackson and has an additional abstract method we need to implement.

**Pull request merged**